src/main/java/org/jenkinsci/plugins/docker/commons/DockerImageExtractor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.docker.commons;
+
+import hudson.ExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.Job;
+import jenkins.model.Jenkins;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+
+/**
+ * Identifies the Docker images used by a Jenkins job. Docker-related plugins can use this
+ * extension point to integrate with for example Docker HUB web hook support.
+ */
+public abstract class DockerImageExtractor implements ExtensionPoint {
```

Comment:
Well, its not a contributor to me, it doesn't contribute to the job, it finds image names that has been configured in a job. I also didn't like the name at first glance, but haven't figured out one that explains it in a similar sense yet, so extractor remained.